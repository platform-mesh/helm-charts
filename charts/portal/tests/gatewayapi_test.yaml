suite: Test Kubernetes Gateway API
values:
- ../values.yaml
release:
  name: platform-mesh
  namespace: platform-mesh-system
chart:
  appVersion: 1.0.0
templates:
  - gatewayapi-httproute.yaml
  - istio-destination-rule.yaml
  - istio-peerauthentication.yaml
  - istio-virtual-service.yaml
tests:
  - it: enable Istio, disable Gateway API - match the Snapshot
    set:
      istio.enabled: true
      gatewayApi.enabled: false
    asserts:
    - matchSnapshot: {}
  - it: disable Istio, enable Gateway API - match the Snapshot
    set:
      istio.enabled: false
      gatewayApi.enabled: true
    asserts:
      - matchSnapshot: {}