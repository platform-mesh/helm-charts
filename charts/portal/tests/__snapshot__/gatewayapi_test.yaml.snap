disable Istio, enable Gateway API - match the Snapshot:
  1: |
    apiVersion: gateway.networking.k8s.io/v1
    kind: HTTPRoute
    metadata:
      name: platform-mesh-portal
    spec:
      hostnames:
        - portal.dev.local
        - '*.portal.dev.local'
      parentRefs:
        - kind: Gateway
          name: k8sapi-gateway
          namespace: default
          sectionName: websecure
      rules:
        - backendRefs:
            - group: ""
              kind: Service
              name: platform-mesh-portal
              namespace: platform-mesh-system
              port: 8080
          matches:
            - path:
                type: PathPrefix
                value: /
enable Istio, disable Gateway API - match the Snapshot:
  1: |
    apiVersion: networking.istio.io/v1alpha3
    kind: DestinationRule
    metadata:
      name: platform-mesh-portal
    spec:
      host: platform-mesh-portal.platform-mesh-system.svc.cluster.local
      trafficPolicy:
        tls:
          mode: ISTIO_MUTUAL
  2: |
    apiVersion: security.istio.io/v1beta1
    kind: PeerAuthentication
    metadata:
      name: platform-mesh-portal
    spec:
      mtls:
        mode: STRICT
      selector:
        matchLabels:
          app: platform-mesh-portal
  3: |
    apiVersion: networking.istio.io/v1alpha3
    kind: VirtualService
    metadata:
      name: platform-mesh-portal
    spec:
      gateways:
        - platform-mesh-system/gateway
      hosts:
        - '*'
      http:
        - corsPolicy:
            allowHeaders:
              - Authorization
              - Content-Type
              - '*'
            allowMethods:
              - GET
              - POST
              - PUT
              - DELETE
          route:
            - destination:
                host: platform-mesh-portal.platform-mesh-system.svc.cluster.local
                port:
                  number: 8080
